pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -229,11 +229,11 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
       LOGGER.info("Table: {} is already balanced", tableNameWithType);
       if (reassignInstances) {
         if (dryRun) {
-          return new RebalanceResult(RebalanceResult.Status.DONE,
+          return new RebalanceResult(RebalanceResult.Status.NO_OP,
```

Comment:
This is not really `NO_OP` because instance might be re-assigned. To improve this, we want to check if the instance re-assignment is also no-op, and return `NO_OP` only if both of the operations are no-op

Reply:
We can consider adding another status, which indicates that assignment is needed, but rebalance is no-op. This way, the caller knows it can be run in sync mode